src/world/terrain/generation/land_generator.cpp

```diff
@@ -143,36 +130,177 @@ JsonToTile::get_volume(glm::imat2x2 center, std::default_random_engine& rand_eng
+
+        std::optional<bool> color_data_b;
+        std::optional<ColorId> color_data_i;
+        std::optional<std::vector<ColorId>> color_data_v;
+        {
+            auto ec_1 = glz::read_json(color_data_b, material_data.color.str);
+
+            if (ec_1) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_1, material_data.color.str)
+                );
+            } else {
+                goto end_parse_color;
+            }
+
+            auto ec_2 = glz::read_json(color_data_i, material_data.color.str);
+
+            if (ec_2) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_2, material_data.color.str)
+                );
+            } else {
+                goto end_parse_color;
+            }
+
+            auto ec_3 = glz::read_json(color_data_v, material_data.color.str);
+
+            if (ec_3) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_3, material_data.color.str)
+                );
+            } else {
+                goto end_parse_color;
+            }
+
+            LOG_CRITICAL(logging::file_io_logger, "could not parse color");
+        }
+
+end_parse_color:
+
+        std::optional<bool> material_data_b;
+        std::optional<MaterialId> material_data_i;
+        std::optional<std::vector<MaterialId>> material_data_v;
+        {
+            auto ec_1 = glz::read_json(material_data_b, material_data.material.str);
+
+            if (ec_1) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_1, material_data.material.str)
+                );
             } else {
-                // add both if it does not
-                materials_w_color.insert({mat_id, {color_id}});
+                goto end_parse_material;
+            }
+
+            auto ec_2 = glz::read_json(material_data_i, material_data.material.str);
+
+            if (ec_2) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_2, material_data.material.str)
+                );
+            } else {
+                goto end_parse_material;
+            }
+
+            auto ec_3 = glz::read_json(material_data_v, material_data.material.str);
+
+            if (ec_3) {
+                LOG_BACKTRACE(
+                    logging::file_io_logger, "Something went wrong {}",
+                    glz::format_error(ec_3, material_data.material.str)
+                );
+            } else {
+                goto end_parse_material;
+            }
+
+            LOG_CRITICAL(logging::file_io_logger, "could not parse material");
+        }
+
+end_parse_material:
```

Comment:
Use a `std::variant`

Reply:
can't put a vetor in a varient

Reply:
https://godbolt.org/z/77a849x6e

Reply:
Ya I realized i was wrong. I was going to do this then realized the next part is... bad...

The spec says no arrays, I remember that as vectors for some reason.

Reply:
Yea arrays makes sense